### ISSUE TITLE
`eslint.nodePath` is not hornored when resolving configurations

### ISSUE THREAD

**Issue by AkatQuas** [microsoft/vscode-eslint#1603]
I have a small project using `eslint` and `vscode-eslint` .

In my case, the totally folder node_modules is located somewhere else.

```bash
|----ROOT
| |----infra
| | |----node_modules
| |----package.json
| |----.eslintrc.js
| |----src
| | |----index.js
```

In vscode terminal, eslint works well with NODE_PATH set.

```bash
NODE_PATH=infra/node_modules infra/node_modules/.bin/eslint src/index.js
```

However, eslint server starts correctly within the `vscode-extension`.

```txt
[Info  - 8:33:55 PM] ESLint library loaded from: /workspace/eslint-demo/infra/node_modules/eslint/lib/api.js
```

however, it couldn't resolve configuration well. Some logs are listed.

```txt
[Error - 8:33:55 PM] An unexpected error occurred:
[Error - 8:33:55 PM] Error: Failed to load config "semistandard" to extend from.
Referenced from: /workspace/eslint-demo/.eslintrc.js
    at configInvalidError (/workspace/eslint-demo/infra/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2627:9)
```

The eslintrc file is listed as the following:

```js
module.exports = {
  extends: [
    'semistandard',
    'standard'
  ]
}
```

Here is my small [project](https://github.com/AkatQuas/eslint-demo). You might need to move node_modules into infra folder manually.

Thanks for the help.

**Comment by MariaSolOs** [comment]
You're using the new configuration file format, so you should set `"eslint.experimental.useFlatConfig": true` in your VS Code settings. You can refer to the `flatConfig` example in the playgrounds folder.

**Comment by AkatQuas** [comment]
Thank you @MariaSolOs . It might be a solution, however it's not possible to *extend* some predefined configs with `flatConfig`. Much more work should be done if we turn into using flat eslint. 

I'm still on my way to sort it out.

**Comment by dbaeumer** [comment]
The underlying reason for this is very likely the working directory used for the ESLint server when starting it. For security reason this is NOT the workspace folder but the installation directory of VS Code. Since you have relative paths in `NODE_PATH` they are very likely not resolved correctly. Can you give it a try and use absolute paths and see if this would make it work.

**Comment by dbaeumer** [comment]
See also https://github.com/microsoft/vscode-eslint/issues/1602

**Comment by AkatQuas** [comment]
It doesn't go well with absolute paths.

In my case, the eslint server does start correctly, but it failed to resolve the configurations.

**Comment by AkatQuas** [comment]
I create a phantom `node_modules` folder which symlink back to `infra/node_modules`. And the extension works well.

Thanks for helping.
